### PR TITLE
Added state machine for creating ai

### PR DESCRIPTION
**Pull request by David-Lam**

### CODE REVIEW

**Review by aaron-lam — APPROVED** [review]

File: Assets/Game/Scripts/InspectorField/TagDropDown/TagSelectorPropertyDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ 
+            if (attrib.UseDefaultTagFieldDrawer)
+            {
+                property.stringValue = EditorGUI.TagField(position, label, property.stringValue);
+            }
+            else
+            {
+                //generate the taglist + custom tags
+                List<string> tagList = new List<string>();
+                tagList.Add("<NoTag>");
```
Comment (makkamchuen) [inline Assets/Game/Scripts/InspectorField/TagDropDown/TagSelectorPropertyDrawer.cs:28]: What GameObject will use the NoTag? And how it is used?

File: Assets/Game/Scripts/PlugableAI/Addon/Sight.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+using System;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "PluggableAI/Addon/Sight"), Serializable]
+public class Sight: ScriptableObject
+{
+  [SerializeField] private float radius;
+  [TagSelector, SerializeField] private string targetTag;
+
+  public bool SearchTarget(StateController controller)
```
Comment (makkamchuen) [inline Assets/Game/Scripts/PlugableAI/Addon/Sight.cs:10]: How does the search logic work? Does it search for player's collider?

**Review by makkamchuen — APPROVED** [review]

File: Assets/Game/Scripts/InspectorField/TagDropDown/TagSelectorPropertyDrawer.cs (hunk 1 above)
Comment (David-Lam) [reply Assets/Game/Scripts/InspectorField/TagDropDown/TagSelectorPropertyDrawer.cs:28]: I copy this code somewhere. `<NoTag>` means empty string or null, we can modify it if need to.

File: Assets/Game/Scripts/PlugableAI/Addon/Sight.cs (hunk 2 above)
Comment (David-Lam) [reply Assets/Game/Scripts/PlugableAI/Addon/Sight.cs:10]: The logic is not decided, right now it just uses Physics.OverlapSphere to search for collider that has the same tag as `targetTag` within `radius` and set the collider's game object to controller's `target`.

**Pull request merged**